### PR TITLE
config.schema.yml: fix permissions schema

### PR DESCRIPTION
**Pull request by mweinelt**
The key reflects a key, value mapping from mxid to "admin", and
additionalProperties requires this to be either a boolean or an object,
but it is currently defined as a list. This breaks validation for us.

> The additionalProperties keyword may be either a boolean or an object

https://json-schema.org/understanding-json-schema/reference/object.html#id2

And objects to me sound like they describe an associative array, or
dictionary:

> Objects are the mapping type in JSON. They map “keys” to “values”.

https://json-schema.org/understanding-json-schema/reference/object.html#object

So let's fix this by replacing the spurious list definition with an
actual object.

### CODE REVIEW

File: config.schema.yml [hunk 1]
```diff
@@ -113,7 +113,7 @@ properties:
                     enableReload:
                         type: "boolean"
                     roomLimit:
-                        type: "number" 
+                        type: "number"
```
Comment (mweinelt) [inline config.schema.yml:116]: drive-by trailing whitespace fix

**Comment by mweinelt** [comment]
I can write a newsblock when we agree this is the correct fix. I am by no means a JSON schema expert.

**Comment by Munksgaard** [comment]
This may be blocking https://github.com/NixOS/nixpkgs/pull/113988

**Review by Half-Shot — APPROVED** [review]
This looks fine to me. Thanks! (Please sign off and newsfile it :+1: )

**Comment by mweinelt** [comment]
This is ready.
